bin/node/src/service.rs

```diff
@@ -183,30 +186,36 @@ pub fn new_partial(config: &Configuration) -> Result<ServiceComponents, ServiceE
     })
 }
 
-#[allow(clippy::type_complexity)]
-#[allow(clippy::too_many_arguments)]
-fn setup(
-    config: Configuration,
-    backend: Arc<FullBackend>,
-    chain_status: SubstrateChainStatus,
-    keystore_container: &KeystoreContainer,
-    import_queue: sc_consensus::DefaultImportQueue<Block>,
-    transaction_pool: Arc<sc_transaction_pool::FullPool<Block, FullClient>>,
-    task_manager: &mut TaskManager,
-    client: Arc<FullClient>,
-    telemetry: &mut Option<Telemetry>,
-    import_justification_tx: mpsc::UnboundedSender<Justification>,
-    collect_extra_debugging_data: bool,
-) -> Result<
-    (
-        RpcHandlers,
-        SubstrateNetwork<Block, BlockHash>,
-        NetworkStarter,
-        SyncOracle,
-        Option<ValidatorAddressCache>,
-    ),
-    ServiceError,
-> {
+struct AlephRuntimeVars {
+    pub session_period: SessionPeriod,
+    pub millisecs_per_block: MillisecsPerBlock,
+}
+
+fn get_aleph_runtime_vars(client: &Arc<FullClient>) -> AlephRuntimeVars {
+    let finalized = client.info().finalized_hash;
+
+    let session_period = SessionPeriod(client.runtime_api().session_period(finalized).unwrap());
+
+    let millisecs_per_block =
+        MillisecsPerBlock(client.runtime_api().millisecs_per_block(finalized).unwrap());
+
+    AlephRuntimeVars {
+        session_period,
+        millisecs_per_block,
+    }
+}
+
+fn get_validator_address_cache(aleph_config: &AlephCli) -> Option<ValidatorAddressCache> {
+    match aleph_config.no_collection_of_extra_debugging_data() {
+        false => Some(ValidatorAddressCache::new()),
```

Comment:
Maybe you want to use [`then_some`](https://doc.rust-lang.org/std/primitive.bool.html#method.then_some)? If you don't like this feel free to just resolve.

Reply:
done

Reply:
Heh, good idea with using `then` instead of `then_some`, but I think the execution is wrong – wouldn't `ValidatorAddressCache::new` work as an argument, without the `|| ` and `()`?

Reply:
Yep, you are like a clippy on this one :)